src/Generator.Rewrite/Extensions.cs

```diff
@@ -0,0 +1,176 @@
+﻿using System;
+using System.Linq;
+using Mono.Cecil;
+
+namespace OpenTK.Rewrite
+{
+    /// <summary>
+    /// Provides convenience extension methods used in the Rewrite namespace.
+    /// </summary>
+    public static class Extensions
```

Comment:
Split this into one extension class per type - MemberReferenceExtensions, CustomAttributeProviderExtensions, etc.